vllm/v1/worker/gpu_model_runner.py

```diff
@@ -774,7 +774,8 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
         # they will be scheduled again sometime in the future.
         scheduled_req_ids = scheduler_output.num_scheduled_tokens.keys()
         cached_req_ids = self.input_batch.req_id_to_index.keys()
-        unscheduled_req_ids = cached_req_ids - scheduled_req_ids
+        resumed_req_ids = scheduler_output.scheduled_cached_reqs.resumed_req_ids
+        unscheduled_req_ids = cached_req_ids - (scheduled_req_ids - resumed_req_ids)
```

Comment:
This is probably worth an explanatory comment - that `cached_req_ids` and `resumed_req_ids` are usually disjoint, apart from the forced-preemption case. And in that case we include the resumed_req_ids in the unscheduled set so that they get cleared from the persistent batch before being re-scheduled in the normal resumed request path.

Reply:
Oh yeah! Great catch.